xwiki-platform-core/xwiki-platform-eventstream/xwiki-platform-eventstream-stores/xwiki-platform-eventstream-store-solr/src/main/java/org/xwiki/eventstream/store/solr/internal/migration/SolrDocumentMigration171000000.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.eventstream.store.solr.internal.migration;
+
+import java.io.IOException;
+import java.util.Date;
+
+import javax.inject.Inject;
+
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.client.solrj.response.QueryResponse;
+import org.apache.solr.common.SolrDocument;
+import org.apache.solr.common.SolrDocumentList;
+import org.apache.solr.common.SolrInputDocument;
+import org.slf4j.Logger;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.eventstream.Event;
+import org.xwiki.eventstream.store.solr.internal.EventsSolrCoreInitializer;
+import org.xwiki.search.solr.SolrException;
+import org.xwiki.search.solr.SolrUtils;
+import org.xwiki.search.solr.XWikiSolrCore;
+
+import jakarta.inject.Singleton;
+
+/**
+ * Migration of documents after introducing {@link Event#FIELD_PREFILTERING_DATE}.
+ *
+ * @version $Id$
+ * @since 17.10.0
+ */
+@Component(roles = SolrDocumentMigration171000000.class)
+@Singleton
+public class SolrDocumentMigration171000000
+{
+    /**
+     * Number of documents to consider at once for migration.
+     */
+    private static final int BATCH_MIGRATION_SIZE = 100;
```

Comment:
Seems a bit low to me for the events core that could have millions of entries. Also, shouldn't the entries be small?

Reply:
I think I took back the batch size from another migration, but yes we should probably increase this value.